torch/_dynamo/replay_record.py

```diff
@@ -40,7 +40,7 @@ def dump(self, f: IO[str]) -> None:
         dill.dump(self, f)
 
     @classmethod
-    def load(cls, f: BinaryIO) -> Self:
+    def load(cls, f: IO[bytes]) -> Self:
```

Comment:
Actually not sure Self is the right typing here for class methods when dealing with subclasses. Need to double check.

Reply:
This is out of scope of this PR